internal/controllers/topology/cluster/desired_state.go

```diff
@@ -175,6 +186,15 @@ func computeControlPlane(_ context.Context, s *scope.Scope, infrastructureMachin
 		return nil, errors.Wrapf(err, "failed to generate the ControlPlane object from the %s", template.GetKind())
 	}
 
+	// Carry over shim owner reference if any.
+	// NOTE: this prevents to the ownerRef to be deleted by server side apply.
+	if s.Current.ControlPlane.Object != nil {
```

Comment:
For non-exusting control planes `s.Current.Controlplane` could be nil.

```suggestion
	if s.Current.ControlPlane != nil && s.Current.ControlPlane.Object != nil {
```